### PR TITLE
Make http options the default configurations

### PR DESCRIPTION
**Pull request by AmatyaAvadhanula**
Makes http configurations the default configurations

### Description

Druid uses Zookeeper dependent options as the default.

Http based options have been tried and tested and are now the recommended options

#### ServerInventoryView : batch -> http

<hr>

<hr>

This PR has:
- [x] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [x] added documentation for new or modified features or behaviors.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [ ] been tested in a test Druid cluster.

### CODE REVIEW

File: services/src/main/java/org/apache/druid/cli/CliOverlord.java [hunk 1]
```diff
@@ -300,7 +300,7 @@ private void configureRunners(Binder binder)
                 binder,
                 "druid.indexer.runner.type",
                 Key.get(TaskRunnerFactory.class),
-                Key.get(ForkingTaskRunnerFactory.class)
+                Key.get(HttpRemoteTaskRunnerFactory.class)
```
Comment (gianm) [inline services/src/main/java/org/apache/druid/cli/CliOverlord.java:303]: With this change, I think there's no need for `druid.indexer.runner.type=httpRemote` in all the bundled configs anymore. Could you try removing that line and see if everything still works OK?

**Review by gianm — COMMENTED** [review]
Concept looks good to me!

File: services/src/main/java/org/apache/druid/cli/CliOverlord.java (hunk 1 above)
Comment (AmatyaAvadhanula) [reply services/src/main/java/org/apache/druid/cli/CliOverlord.java:303]: Done, thanks!

**Comment by cryptoe** [comment]
How will the upgrade look when people go from zk-based task management to HTTP-based task management considering most of the people running their clusters would be using default values? 

IMHO we should tag release notes on this PR as well.

**Comment by kfaraz** [comment]
Upgrade should not be a concern for the loadqueue peon type and the server inventory view mode being changed to "http" as these take effect when the coordinator is restarted. The respective endpoints are already exposed by the historical, so the change should be seamless.

Task management should be fine too if the upgrade order is maintained i.e. MM/indexer before overlord. An upgraded middle manager would be listening on HTTP for new tasks but the overlord would start sending HTTP task assignments only when it is upgraded too.

cc: @cryptoe

**Comment by kfaraz** [comment]
@AmatyaAvadhanula , there seems to be some coverage failure. I think it can be remedied if you add a line to verify the loadqueue peon type in `DruidCoordinatorConfigTest`.

**Review by kfaraz — APPROVED** [review]
LGTM

**Pull request merged**